src/utils.cpp

```diff
@@ -562,6 +562,31 @@ std::string utils::get_potential_date(const std::string & path, const std::strin
   return "";
 }
 
+/* ----------------------------------------------------------------------
+   read first line of potential file
+   if it has UNITS field, return following word
+------------------------------------------------------------------------- */
+
+std::string utils::get_potential_units(const std::string & path, const std::string & potential_name) {
+  TextFileReader reader(path, potential_name);
+  reader.ignore_comments = false;
+  char * line = nullptr;
+
+  while ((line = reader.next_line())) {
```

Comment:
This looks like it is going to read the whole file, not just the first line.  Apart from performance, it might lead to false matches. Why not just limit it to the first line?

Reply:
That is a valid observation. I copied and adapted the function for reading the "DATE:" tag, so the same issue applies to that one as well. Just made the change. Thanks.